### PR TITLE
[RPGF-41] Fix CRON Job

### PR DESCRIPTION
**Pull request by mmxxixx**
# Description

[//]: # "Please include a summary of the changes and the related issue. (You can write below without deleting this comment)"

### What
Fix cronjob from not schedule the job, add cors base on env and change some part of code to use lib

[//]: # "Briefly describe this feature / fix / refactor (You can write below without deleting this comment)"

### Why
[//]: # "Describe the reason why we need this feature / fix / refactor or refer to Jira Ticket (You can write below without deleting this comment)"

## Type of change

Please check the relevent option with `x`.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if needed)
- [ ] My changes generate no new warnings
- [X] I have tested on my `Local Machine`

### CODE REVIEW

File: lib/save-file/save-file.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            } catch (mkdirError) {
+                console.error("Error creating directory:", mkdirError);
+                throw new Error("Failed to create directory. Check permissions and path.");
+            }
+        }
+    
+        const filePath = path.join(folderName, fileName);
+        
+        // Use fs.writeFileSync for more robust file writing
+        fs.writeFile(filePath, data, {encoding:"utf8", flag:"w"}, () => {});
```
Comment (Billy19191) [inline lib/save-file/save-file.ts]: Why we need () => {}?, can we delete it?

File: lib/save-file/save-file.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        console.log(`File successfully written to ${filePath}`);
+    } catch (error) {
+        console.error("An error occurred during the save file process:", error);
+        throw error; // Re-throw the error for proper handling upstream
+    }
+}
```
[the remark this answers is not in the document]
Comment (mmxxixx) [reply lib/save-file/save-file.ts]: writeFile is asynchronous, while writeFileSync is synchronous
so I will fix it to writeFileSync then

**Review by Billy19191 — APPROVED** [review]
Nothing change on my side 💯

**Comment by wit03** [comment]
@TGRZiminiar yung tong merge mai na

**Review by wit03 — APPROVED** [review]
LGTV+

**Pull request merged**